projects/plugins/jetpack/package.json

```diff
@@ -140,13 +140,13 @@
 		"@babel/register": "7.17.7",
 		"@babel/runtime": "7.17.9",
 		"@size-limit/preset-app": "5.0.5",
-		"@testing-library/dom": "7.31.2",
+		"@testing-library/dom": "8.13.0",
 		"@testing-library/jest-dom": "5.14.1",
 		"@testing-library/preact": "2.0.1",
-		"@testing-library/react": "11.2.7",
-		"@testing-library/react-hooks": "4.0.1",
-		"@testing-library/user-event": "12.8.3",
-		"@types/react": "18.0.8",
+		"@testing-library/react": "12.1.5",
+		"@testing-library/react-hooks": "8.0.0",
+		"@testing-library/user-event": "14.1.1",
+		"@types/react": "17.0.44",
```

Comment:
@anomiex because of this downgrade of `@types/react`, we now have conflicts between two versions (v17 and v18). This causes a few TypeScript errors. Please see this discussion - https://github.com/Automattic/jetpack/pull/24256#discussion_r866101448

I know `@testing-library/react-hooks` has a v17 peer dependency of `@types/react`, can we do this in `.pnpmfile.cjs` to fix the Peer dependencies for `@testing-library/react-hooks`? In v8, `@testing-library/react-hooks` has made those type peer deps optional, so I think we can override the versions.


```diff
@@ -137,6 +137,10 @@ function fixPeerDeps( pkg ) {
                pkg.peerDependencies[ 'js-git' ] = '*';
        }
 
+       if ( pkg.name === '@testing-library/react-hooks' && pkg.peerDependencies?.[ '@types/react' ] ) {
+               pkg.peerDependencies[ '@types/react' ] = '*';
+       }
+
```

CC @Automattic/jetpack-agora

Reply:
Replied on your PR at #24294